### PR TITLE
Feature/add month year question step

### PR DESCRIPTION
**Pull request by mauriciosouto**
### Description

Update Date Question to support `MM/yyyy` format 
A new DatePickerDialog was added to support two spinners selection only for Month and Year since the Android calendar doesn't support that.
Fix issue with consent and soft keyboard on Consent

### Acceptance Criterial

- [ ] Open app, take Disease History survey, check that step 3 is a MonthYear Date question
- [ ] Save resultas after select a MontYear date, retake the survey and make sure the selected value is shown correctly
- [ ] Start Registration process, check that the keyboard is closed after the sign information is entered and the user press "Next"

### CODE REVIEW

File: backbone/src/main/java/org/researchstack/backbone/ui/step/body/DateQuestionBody.java [hunk 1]
```diff
@@ -61,6 +69,15 @@ public DateQuestionBody(Step step, StepResult result) {
                 savedTimeInMillis = ((Double) this.result.getResult()).longValue();
             } else if (this.result.getResult() instanceof Long) {
                 savedTimeInMillis = (Long) this.result.getResult();
+            } else if (this.result.getResult() instanceof String) {
+                try {
+                    SimpleDateFormat parseFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+                    Date parsedResult = parseFormat.parse((String)this.result.getResult());
```
Comment (peterTarmac) [inline backbone/src/main/java/org/researchstack/backbone/ui/step/body/DateQuestionBody.java]: There's a space missing before **this**

File: backbone/src/main/java/org/researchstack/backbone/ui/step/layout/FormStepLayout.java [hunk 2]
```diff
@@ -286,8 +288,19 @@ protected void onNextClicked()
...
+                imm.hideSoftInputFromWindow(container.getWindowToken(), 0);
+            }
+            final Handler handler = new Handler();
+            handler.postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    updateAllQuestionSteps(false);
+                    callbacks.onSaveStep(StepCallbacks.ACTION_NEXT, formStep, stepResult);
+                }
+            }, 100);
```
Comment (peterTarmac) [inline backbone/src/main/java/org/researchstack/backbone/ui/step/layout/FormStepLayout.java:303]: Wasn't this already in somewhere else?

File: backbone/src/main/java/org/researchstack/backbone/ui/views/MonthYearPickerDialog.java [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    private Date mMaxDate = null;
+    private int mMaxYear = -1;
+    private int mMaxMonth = 12;
+
+    private Date mMinDate = null;
+    private int mMinYear = 1900;
+    private int mMinMonth = 1;
+
+    private MonthYearPickerDialog.OnDateSetListener mDateSetListener;
+    private View dialogView;
```
Comment (peterTarmac) [inline backbone/src/main/java/org/researchstack/backbone/ui/views/MonthYearPickerDialog.java]: mDialogView

**Comment by peterTarmac** [comment]
Just a few style comments and/or questions, other than that looks good 👍

File: backbone/src/main/java/org/researchstack/backbone/ui/step/layout/FormStepLayout.java (hunk 2 above)
Comment (mauriciosouto) [reply backbone/src/main/java/org/researchstack/backbone/ui/step/layout/FormStepLayout.java:303]: It is, and testing the app I found was needed on another part of the app :(

**Pull request merged**